### ISSUE TITLE
Stickies sometimes shoot through corners

### ISSUE THREAD

**Issue by hexaflexahexagon** [momentum-mod/game#1394]
**Describe the bug**
Stickies sometimes seem to shoot straight through a corner, see the videos below for examples. The bsp and vmf can both be made available to whoever chooses to debug this but I won't post them here because the map pictured below is not yet public.

Seems to be possibly related to https://github.com/momentum-mod/game/issues/1350? I noticed stickybombs "jumping" pretty far while trying to find an angle for this bug but that could be unrelated.

```
setpos_exact -5509.918457 -7177.558105 8601.253906;setang_exact 61.155285 28.953585 0.000000
```

mmod:

https://user-images.githubusercontent.com/5162166/127766190-7942c87e-9a74-4d75-a4f5-231fa1dbda10.mp4

tf2:

https://user-images.githubusercontent.com/5162166/127766203-f4d0e0b4-c85c-4f32-afff-611f40b9844f.mp4

The area is properly sealed in hammer:
![image](https://user-images.githubusercontent.com/62961532/123537526-e14e3100-d6fd-11eb-9f7c-39dd645aa016.png)

**To Reproduce**
1. Load the map pictured in those videos
2. Enter practice mode and then use the `setpos_exact` command above.
3. Shoot stickybomb at magic angle
4. Bomb vanishes

**Desktop (*please complete the following information*):**
OS: WIndows
Renderer: Dx9
UI: VGUI

**Additional context**
Reported by reero in discord

**Comment by hexaflexahexagon** [comment]
i made another test map and found consistent setups:
download: https://cdn.discordapp.com/attachments/671175695648423959/864987333664440401/sj_anotherstickybugs.bsp

The "e" bind will use a corner made of 1 unit thick walls, while the "r" bind uses one made of 32 unit thick walls. Both have the same effect.
`bind e "setpos_exact 422.506866 389.444366 340.031250;setang_exact 47.982414 12.515610 0.000000"` 
`bind r "setpos_exact -350.224609 457.165161 352.031250;setang_exact 46.398766 -147.049713 0.000000"`

I used a macro (mouse press, 100ms delay, mouse release) and a `host_timescale 10` to find these. It looks like they don't work great on `host_timescale 1` but i'm pretty sure that's unrelated (maybe due to 100ms being enough to charge slightly at 10x speed idk) and finding these at normal speed is pretty annoying due to how slow shooting stickybombs is. 

e: and some things to note about this bug that may help diagnosis. 
1. when this happens all of the particles instantly disappear, as if the stickybomb was completely deleted / unrendered completely rather than just flying somewhere else. 
2. the stickybomb still deals knockback when this happens and it is invisible. this is especially peculiar considering stickybombs don't deal damage through walls typically

**Comment by braem** [comment]
Had a similar thing happen to me on stream, though it was not near any wall: https://www.twitch.tv/braemie/clip/OpenHelpfulNikudonTBCheesePull-WJZi-4o7m2g95DLT

Fun

**Comment by hexaflexahexagon** [comment]
there is a brush seam there so im guessing it's related if not the same? would need to investigate more

**Comment by jason-e** [comment]
This bug doesn't seem to happen when using hex's example map and locations -- is it fixed somehow?

Also, is this normal? Stickies seem to be able to partially enter walls/floors sometimes when shooting them in corners.

https://user-images.githubusercontent.com/2745352/170812339-ea2071ac-b270-484d-a4e9-54b514544727.mp4

**Comment by hexaflexahexagon** [comment]
No, that is not normal and doesn't happen in tf2 (tho I remember it happening when making this test setup map so not new). It's almost like the sticky is colliding with 1 plane and then somehow snapping its position along that plane, ignoring the nearby surface that it should also be colliding with. 

It's possible the original bug was the same as this bug but just much more extreme (going far enough to make it invisible instantly) and some fix happened to slightly improve this at some point.
